### PR TITLE
DOCSUP-6113: Document the Atomic database engine

### PR DESCRIPTION
**Pull request by adevyatova**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Documentation for #12343, #15448

...

By adding documentation, you'll allow users to try your new feature immediately, not when someone else will have time to document it later. Documentation is necessary for all features that affect user experience in any way. You can add brief documentation draft above, or add documentation right into your patch as Markdown files in [docs](https://github.com/ClickHouse/ClickHouse/tree/master/docs) folder.

If you are doing this for the first time, it's recommended to read the lightweight [Contributing to ClickHouse Documentation](https://github.com/ClickHouse/ClickHouse/tree/master/docs/README.md) guide first.

Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by tavplubix** [comment]
Please, do not change anything related to "submodules" and "contrib". If `git` tells you that some submodules are modified after merging with `master` and you see something like
```
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

  modified:   contrib/libpq (new commits)
  modified:   contrib/replxx (new commits)
  modified:   contrib/simdjson (new commits)
```
then just execute
```
git submodule init && git submodule sync && git submodule update
```
If you still see some changes in submodules after the command above, then you can try the ultimate submodule-combo:
```
sudo rm -rf contrib/zlib-ng/CMakeFiles/CMakeTmp contrib/avro/lang/c++/include/avro && git submodule sync --recursive && git submodule update --init --recursive && git submodule foreach git reset --hard && git submodule foreach git clean -xfd && git submodule foreach git submodule sync --recursive && git submodule foreach git submodule update --init --recursive && git submodule foreach git submodule foreach git reset --hard && git submodule foreach git submodule foreach git clean -xfd
```

Or ask someone for help if something strange is happening with git.

File: docs/en/engines/database-engines/atomic.md [hunk 1]
```diff
@@ -3,15 +3,60 @@ toc_priority: 32
...
+### DROP TABLE {#drop-table}
+
+On `DROP TABLE` no data is removed, database `Atomic` just marks table as dropped by moving metadata to `/clickhouse_path/metadata_dropped/` and notifies background thread. Running queries still may use dropped table. Table will be actually removed when it's not in use.
+
+### RENAME TABLE {#rename-table}
+
+`RENAME` queries are performed without changing UUID and moving table data. 
+
+!!! note "Note"
+    `RENAME` and `DROP` are execute without IStorage-level RWLocks. These queries do not wait for the completion of queries using the table and will be executed instantly. Data deletion occurs asynchronously in a background thread.
```
Comment (tavplubix) [inline docs/en/engines/database-engines/atomic.md:36]:
```suggestion
`RENAME` queries are performed without changing UUID and moving table data. These queries do not wait for the completion of queries using the table and will be executed instantly.
```

File: docs/en/engines/database-engines/atomic.md [hunk 2]
```diff
@@ -3,15 +3,60 @@ toc_priority: 32
...
+
+### DELETE/DETACH {#delete-detach}
```
Comment (tavplubix) [inline docs/en/engines/database-engines/atomic.md:38]:
```suggestion
### DROP/DETACH {#drop-detach}
```

There is no `DELETE` query

File: docs/en/engines/database-engines/atomic.md [hunk 3]
```diff
@@ -3,15 +3,60 @@ toc_priority: 32
...
+
+`DELETE` and `DETACH` queries are executed asynchronously — waits for the running `SELECT` queries to finish but is invisible to the new queries.
```
Comment (tavplubix) [inline docs/en/engines/database-engines/atomic.md:40]: If synchronous mode is not enabled by `SYNC` modifier of `database_atomic_wait_for_drop_and_detach_synchronously` setting, then it does not wait for running `SELECT`s, `INSERT`s and other queries which are using the table

File: docs/en/operations/server-configuration-parameters/settings.md [hunk 4]
```diff
@@ -125,6 +125,25 @@ Settings profiles are located in the file specified in the parameter `user_confi
...
+<default_replica_path>/clickhouse/tables/{uuid}/{shard}</default_replica_path>
+```
+## default_replica_name {#default_replica_name}
+
+ The replica name in ZooKeeper.
+
+**Example**
+
+``` xml
+<default_replica_name>/clickhouse/tables/{uuid}/{replica}</default_replica_name>
```
Comment (tavplubix) [inline docs/en/operations/server-configuration-parameters/settings.md:144]:
```suggestion
<default_replica_name>{replica}</default_replica_name>
```

File: docs/en/operations/settings/settings.md [hunk 5]
```diff
@@ -2747,4 +2747,36 @@ Possible values:
 
 Default value: `0`.
 
+## database_atomic_delay_before_drop_table_sec {#database_atomic_delay_before_drop_table_sec}
```
Comment (tavplubix) [inline docs/en/operations/settings/settings.md:2750]: It's server configuration parameter

**Comment by adevyatova** [comment]
@tavplubix please, review the fixes!

File: docs/en/operations/server-configuration-parameters/settings.md [hunk 6]
```diff
@@ -100,6 +100,11 @@ Default value: `1073741824` (1 GB).
     <size_limit>1073741824</size_limit>
 </core_dump> 
 ```
+## database_atomic_delay_before_drop_table_sec {#database_atomic_delay_before_drop_table_sec}
+
+Sets the delay before remove table data in seconds. If the query has `SYNC` modifier, this setting is ignore.
```
Comment (tavplubix) [inline docs/en/operations/server-configuration-parameters/settings.md:105]:
```suggestion
Sets the delay before remove table data in seconds. If the query has `SYNC` modifier, this setting is ignored.
```

**Review by tavplubix — COMMENTED** [review]
English version LGTM

File: docs/ru/engines/database-engines/atomic.md [hunk 7]
```diff
@@ -3,15 +3,52 @@ toc_priority: 32
...
+``` sql
+    CREATE DATABASE test[ ENGINE = Atomic];
+```
+
+## Особенности и рекомендации {#specifics-and-recommendations}
+
+### UUID {#table-uuid}
+
+Каждая таблица в базе данных `Atomic` имеет уникальный [UUID](../../sql-reference/data-types/uuid.md) и хранит данные в папке `/clickhouse_path/store/xxx/xxxyyyyy-yyyy-yyyy-yyyy-yyyyyyyyyyyy/`, где `xxxyyyyy-yyyy-yyyy-yyyy-yyyyyyyyyyyy` - это UUID таблицы.
+Обычно UUID генерируется автоматически, но пользователь также может явно указать UUID в момент создания таблицы (однако, это не рекомендуется). Для отображения UUID в запросе `SHOW CREATE` вы можете использовать настройку [show_table_uuid_in_table_create_query_if_not_nil](../../operations/settings/settings.md#show_table_uuid_in_table_create_query_if_not_nil). Результат выполнения в таком случае будет иметь вид:
```
Comment (olgarev) [inline docs/ru/engines/database-engines/atomic.md:21]:
```suggestion
Обычно UUID генерируется автоматически, но пользователь также может явно указать UUID в момент создания таблицы (однако это не рекомендуется). Для отображения UUID в запросе `SHOW CREATE` вы можете использовать настройку [show_table_uuid_in_table_create_query_if_not_nil](../../operations/settings/settings.md#show_table_uuid_in_table_create_query_if_not_nil). Результат выполнения в таком случае будет иметь вид:
```

File: docs/ru/engines/database-engines/atomic.md [hunk 8]
```diff
@@ -3,15 +3,52 @@ toc_priority: 32
...
 ```
+### RENAME TABLE {#rename-table}
+
+Запросы `RENAME` выполняются без изменения UUID и перемещения табличных данных. Эти запросы не ожидают завершения использующих таблицу запросов и будут выполнены мгновенно. 
+
+### DROP/DETACH TABLE {#drop-detach-table}
+
+При выполнении запроса `DROP TABLE` никакие данные не удаляются. Таблица помечается как удаленная, метаданные перемещаются в папку `/clickhouse_path/metadata_dropped/` и база данных уведомляет фоновый поток. Задержка перед окончательным удалением данных задается настройкой [database_atomic_delay_before_drop_table_sec](../../operations/server-configuration-parameters/settings.md#database_atomic_delay_before_drop_table_sec).
+Вы можете задать синхронный режим определяя модификатор `SYNC`. Используйте для этого настройку [database_atomic_wait_for_drop_and_detach_synchronously](../../operations/settings/settings.md#database_atomic_wait_for_drop_and_detach_synchronously). В этом случае запрос `DROP` ждет завершения `SELECT`, `INSERT` и других запросов, которые используют таблицу. Таблица будет фактически удалена, когда она не будет использоваться.
```
Comment (olgarev) [inline docs/ru/engines/database-engines/atomic.md:33]:
```suggestion
Вы можете задать синхронный режим, определяя модификатор `SYNC`. Используйте для этого настройку [database_atomic_wait_for_drop_and_detach_synchronously](../../operations/settings/settings.md#database_atomic_wait_for_drop_and_detach_synchronously). В этом случае запрос `DROP` ждет завершения `SELECT`, `INSERT` и других запросов, которые используют таблицу. Таблица будет фактически удалена, когда она не будет использоваться.
```

**Review by tavplubix — APPROVED** [review]

**Pull request merged**